### PR TITLE
Add documentation on how to create AWS backed workers

### LINKED ISSUE #4741: Add documentation on how to do a basic AWS AMI build with monopacker to use with your taskcluster instance

### PR DESCRIPTION
**Pull request by MasterWayZ**
Wrote this from as best as I could remember, looking at all the files I've edited and the process I went through.

Github Bug/Issue: Fixes #4741

### CODE REVIEW

File: dev-docs/creating-a-worker-ami.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+### Configuring the template
+
+First we edit `template/builders/amazon_ebs.jinja2` and under `ami_users` set your AWS account ID.
+
+Then we edit `template/vars/default_aws.yaml` and we change the following:
+
+* Change `instance_type` to an instance type that has the amount of capacity that you will need. For example `c5.large`.
+* Change `region:` to the region that you want to build the AMI in.
+* Change `ami_regions:` to regions where you wish to copy the AMI to and use it to deploy workers.
+* Change `volume_size:` to the volume size that you want your boot volume to be, in GB.
```
Comment (djmitche) [inline dev-docs/creating-a-worker-ami.md:31]: Did you end up increasing or decreasing this?  Any particular reason to do so?

**Review by djmitche — COMMENTED** [review]
This pile of comments looks huge but really, this is great and I think will be a big help to the next person to attempt this.

File: dev-docs/creating-a-worker-ami.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+To install everything we need, from within the venv, run `python setup.py install`.
+
+### Configuring the template
+
+First we edit `template/builders/amazon_ebs.jinja2` and under `ami_users` set your AWS account ID.
+
+Then we edit `template/vars/default_aws.yaml` and we change the following:
+
+* Change `instance_type` to an instance type that has the amount of capacity that you will need. For example `c5.large`.
```
[the remark this answers is not in the document]
Comment (MasterWayZ) [reply dev-docs/creating-a-worker-ami.md:28]: I think it would depend on what jobs you run.

File: dev-docs/creating-a-worker-ami.md (hunk 1 above)
Comment (MasterWayZ) [reply dev-docs/creating-a-worker-ami.md:31]: I changed the ami_regions to only be one region (for cost savings), same for volume_size, I made it 20GB, for cost savings.

File: dev-docs/creating-a-worker-ami.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+providers: {
+    "provider-name": {
+      "providerType": "aws",
+      "credentials": {
+        "accessKeyId": "accesskey",
+        "secretAccessKey": "secretkey"
+      }
+    }
+  }
+```
```
[the remark this answers is not in the document]
Comment (MasterWayZ) [reply dev-docs/creating-a-worker-ami.md:83]: Like adding the link or removing it and only having the link?

File: dev-docs/creating-a-worker-ami.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  - default_linux
+  - default_aws
+  - default_firefoxci
+  - amazon_ebs_bionic
+
+script_directories:
+  - ubuntu-bionic
+  - worker-runner-linux
+  - docker-worker-linux
+```
```
[the remark this answers is not in the document]
Comment (MasterWayZ) [reply dev-docs/creating-a-worker-ami.md:54]: Okay, not sure how that works though.

**Comment by petemoore** [comment]
This looks awesome! Thanks Michelle. I'll let Dustin take it from here.

File: dev-docs/creating-a-worker-ami.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply dev-docs/creating-a-worker-ami.md:28]: It wouldn't -- this only sets the instance size that monopacker uses to build the image.  The resulting image can be used on a range of sizes.

File: dev-docs/creating-a-worker-ami.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply dev-docs/creating-a-worker-ami.md:54]: I think `monopacker packer-template mynewbuilder` can help experiment with this a bit.

File: dev-docs/creating-a-worker-ami.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (djmitche) [reply dev-docs/creating-a-worker-ami.md:85]: An example plus a link to the full docs is, I think, the most helpful approach.

File: dev-docs/creating-a-worker-ami.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (MasterWayZ) [reply dev-docs/creating-a-worker-ami.md:28]: Oh right!

[the remark this answers is not in the document]
Comment (djmitche) [reply dev-docs/creating-a-worker-ami.md:28]: (in fact that's easy to misunderstand -- perhaps a comment to remind readers would be good)

**Review by djmitche — APPROVED** [review]
🥳

**Pull request merged**